### DISCUSSION TITLE
Optimize initial lens model for DESIJ1225-0725

### DISCUSSION THREAD

**Discussion by SulymanHossain** [AstroBridge/BDLensing#130] [Q&A]
I was trying to optimize the DESIJ1225-0725 model by changing the parameter's value and masking choice. The lowest reduced Chi-square value I got is -1.395 with 50 iterations and 50 particles. Here is my Colab [Notebook](https://drive.google.com/file/d/1C3nKMYPLIdHYsvssTgimShRkU6cPD_Bv/view?usp=sharing) and other fitted results are

**Lens Mass Results**: 
[{'theta_E': 1.5018471726490392, 'gamma': 2.405905053259318, 'e1': 0.033852417117589795, 'e2': 0.03523727464449423, 'center_x': -0.8059045691250906, 'center_y': 0.45061411891890313}, {'gamma1': 0.08105673455670577, 'gamma2': 0.13634542633116387, 'ra_0': 0.0, 'dec_0': 0.0}]

**Source Result**: [{'amp': 1, 'R_sersic': 0.13411163376418153, 'n_sersic': 1.0, 'e1': -0.008336457139475624, 'e2': -0.087904011109995, 'center_x': 0.10516146262252192, 'center_y': 0.14930378096857358}, {'amp': 1, 'n_max': 11, 'beta': 0.3220962534795221, 'center_x': 0.10516146262252192, 'center_y': 0.14930378096857358}]

**Lens Light Result**:   [{'amp': 1, 'R_sersic': 1.015575434108388, 'n_sersic': 4.0, 'e1': 0.033852417117589795, 'e2': 0.03523727464449423, 'center_x': -0.8059045691250906, 'center_y': 0.45061411891890313}]

![d1](https://github.com/ajshajib/BDLensing/assets/132705545/5c645837-4e1a-464f-b20e-cdf966a79623)
![d2](https://github.com/ajshajib/BDLensing/assets/132705545/44139df7-064e-4f1c-9ab4-398e7c5dac76)
If we observe the Chi-square map, there are some strong residuals at the upper side of the lens galaxy. And most of my fit gave the same effect. From this what I understood is, Lenstronomy is trying to find a counter image at the upper side of the lens center for the unmasked lights. I don't know why it is not fitting those two unmasked lights as each other's mirror. Is this because my model isn't just good enough? Or it is simply not a lensing system? Please share your thoughts.

**Comment by rafee-adnan** [comment]
The following suggestions may help with your fitting:
- Your gamma value is high. Try fixing it to 2, or you can set the limits close to 2.
- The shapelets are trying to recreate the lens light around the center. You can add another sersic profile to your `lens_light_model` to get rid of this problem.
- Don't take `n_max` beyond 10. Your lens will fit good around 8. If you find the initial fit, then you can dial it up.

**Reply by SulymanHossain** [reply]
> The following suggestions may help with your fitting:
> 
> * Your gamma value is high. Try fixing it to 2, or you can set the limits close to 2.
> * The shapelets are trying to recreate the lens light around the center. You can add another sersic profile to your `lens_light_model` to get rid of this problem.
> * Don't take `n_max` beyond 10. Your lens will fit good around 8. If you find the initial fit, then you can dial it up.

After fixing `gamma=2` and adding a double Sersic light profile to `lens_light_model` it did significantly reduce the Reduced Chi-square value and the residuals around the center have improved.
`n_max=8`

![26](https://github.com/ajshajib/BDLensing/assets/132705545/e91fffbf-c136-411e-9d40-3529c7a886ae)
![26](https://github.com/ajshajib/BDLensing/assets/132705545/21926582-8ce9-40a6-9245-f62093809d0c)
![26+](https://github.com/ajshajib/BDLensing/assets/132705545/58a293b0-2ee6-4cfe-a194-ebe066d7e4ce)


`n_max=9`

![27](https://github.com/ajshajib/BDLensing/assets/132705545/0ac40da2-c5e4-4696-85ae-9c4dad0a2016)
![27](https://github.com/ajshajib/BDLensing/assets/132705545/6a2d84bf-d384-4d89-bfb8-1af25c573e64)

`n_max=10`

![28](https://github.com/ajshajib/BDLensing/assets/132705545/f04cdc4a-0ede-4b4a-b002-e7d71f25ce6d)
![28](https://github.com/ajshajib/BDLensing/assets/132705545/d882834b-8a13-4921-9cc7-a730b6d87b49)

 Do you have any suggestions to further reduce the residuals near center?

**Reply by rafee-adnan** [reply]
Seeing the pso fitting of all the results above your beta value and your R_sersic are still higher than usual. Try constraining it to a lower value. You can try setting the upper limit to something like 0.2” for both R_sersic and beta.

**Reply by SulymanHossain** [reply]
I tried some combinations where I set the upper limit of beta as 0.1, 0.15, 0.2, etc. But It introduces more residuals in the plot which increased the reduced Chi-square value above the permitted limit. I don't know what to do now.

**Reply by rafee-adnan** [reply]
Share the latest plotted result, your decomposition plot

**Reply by SulymanHossain** [reply]
Here is the results.
-1.4226570085760235 reduced X^2 of best position

**Lens Mass result**: [{'theta_E': 2.492040185843647, 'gamma': 2, 'e1': -0.4358026979625224, 'e2': 0.20854333139945955, 'center_x': -0.6333592660091675, 'center_y': -0.3315648385788836}, {'gamma1': -0.24937447502371235, 'gamma2': 0.1033446057475215, 'ra_0': -0.37428105996536154, 'dec_0': 0.2042304127334315}]

**Source result**: [{'amp': 1, 'R_sersic': 0.14078299035764694, 'n_sersic': 1.0, 'e1': 0.1172547746163036, 'e2': -0.4049207526280878, 'center_x': 0.08478946850121045, 'center_y': 0.0002857913396861209}, {'amp': 1, 'n_max': 8, 'beta': 0.11418158334194975, 'center_x': 0.08478946850121045, 'center_y': 0.0002857913396861209}]

**Lens Light result**: [{'amp': 1, 'R_sersic': 1.7653881335932176, 'n_sersic': 1.0, 'e1': -0.05935995308867987, 'e2': -0.36661187795904177, 'center_x': -0.8066313752969785, 'center_y': 0.4508280265833587}, {'amp': 1, 'R_sersic': 1.0227948956496624, 'n_sersic': 4.0, 'e1': 0.06711865352892302, 'e2': 0.0266083156529855, 'center_x': -0.8066313752969785, 'center_y': 0.4508280265833587}]

![d3](https://github.com/ajshajib/BDLensing/assets/132705545/134b19f3-9a7e-4b45-8bdf-437cda27f875)
![d4](https://github.com/ajshajib/BDLensing/assets/132705545/14cc1e6c-9f02-48c6-b9d7-a3baeb87bb47)
